ultraviolet/shared/src/main/scala/ultraviolet/syntax.scala

```diff
@@ -91,4 +92,68 @@ object syntax extends ShaderDSLOps:
   def PrecisionMediumPFloat: ShaderHeader = ShaderHeader.PrecisionMediumPFloat
   def PrecisionLowPFloat: ShaderHeader    = ShaderHeader.PrecisionLowPFloat
 
+  private[ultraviolet] object interpolators:
+
+    object hex:
+      private val hexGroup: String = "([0-9A-F]{2})"
+      private val hex3: Regex      = List.fill(3)(hexGroup).mkString("(?i)#", "", "").r
+      private val hex4: Regex      = List.fill(4)(hexGroup).mkString("(?i)#", "", "").r
+
+      private def toScaledFloat(string: String): Float = Integer.parseInt(string, 16) / 255f
+
+      extension (string: String) {
+        def toVec3: Option[vec3] = Option(string).collect { case hex3(r, g, b) =>
+          vec3(toScaledFloat(r), toScaledFloat(g), toScaledFloat(b))
+        }
+
+        def toVec4: Option[vec4] = Option(string).collect { case hex4(r, g, b, a) =>
+          vec4(toScaledFloat(r), toScaledFloat(g), toScaledFloat(b), toScaledFloat(a))
+        }
+      }
+
+    object rgb:
+      private def is8bit(i: Int): Boolean                      = i >= 0 && i < 256
+      private def toScaledFloat(string: String): Option[Float] = string.toIntOption.filter(is8bit).map(_ / 255f)
+
+      extension (string: String) {
+        def toVec3: Option[vec3] = Option(string.split(",").toList.map(toScaledFloat)).collect {
+          case Some(r) :: Some(g) :: Some(b) :: Nil => vec3(r, g, b)
+        }
+
+        def toVec4: Option[vec4] = Option(string.split(",").toList.map(toScaledFloat)).collect {
+          case Some(r) :: Some(g) :: Some(b) :: Some(a) :: Nil => vec4(r, g, b, a)
+        }
+      }
+
+  extension (sc: StringContext) {
+
+    @SuppressWarnings(Array("scalafix:DisableSyntax.throw"))
+    def hex(args: Any*): vec3 =
+      import interpolators.hex.*
+      sc.s(args*).toVec3.getOrElse {
+        throw IllegalArgumentException(s"Invalid hex triplet ${args.mkString} (expected format '#00ff00')")
+      }
+
+    @SuppressWarnings(Array("scalafix:DisableSyntax.throw"))
+    def hexa(args: Any*): vec4 =
+      import interpolators.hex.*
+      sc.s(args*).toVec4.getOrElse {
+        throw IllegalArgumentException(s"Invalid hex triplet ${args.mkString} (expected format '#00ff00ff')")
```

Comment:
Apparently hex triplet can also refer to a quadruplet, [according to wikipedia](https://en.wikipedia.org/wiki/Web_colors#Hex_triplet). Let me know if this message could do with a review.

Reply:
I don't think there's any problem here, but similar to my other comment, I might go with something like:

```
s"Invalid hex triplet/value ${args.mkString}. Supported formats are '#00ff00' or '#00ff00ff', using the `hex` and `hexa` interpolators, respectively."
```

Reply:
Ok, will do. I also like your suggestion of `values` instead of `triplets` - I'll go with that.
The error messages have different formats to the ones in `CreateShaderAST`, but maybe that's ok.
Let me know what you think.